src/wallet/test/fuzz/spend.cpp

```diff
@@ -0,0 +1,254 @@
+    CWallet& wallet = WalletSingleton::GetInstance().GetWallet();
+    CoinsResult& available_coins = WalletSingleton::GetInstance().GetCoins();
+
+    // Asserting if the `wallet` has no funds.
+    std::vector<COutput> coins = available_coins.All();
+    assert(!coins.empty());
+
+    CCoinControl coin_control;
+    CMutableTransaction mtx;
+
+    LIMITED_WHILE(fuzzed_data_provider.ConsumeBool(), 505)
+    {
+        CallOneOf(
+            fuzzed_data_provider,
+            [&] {
+                // Creates a new `CoinControl`.
+                coin_control = GetNewCoinControl(fuzzed_data_provider, wallet, coins);
+            },
+            [&] {
+                // Creates a new Transaction using `CreateTransaction`.
+                std::vector<CRecipient> recipients;
+                unsigned int recipients_size = fuzzed_data_provider.ConsumeIntegralInRange<unsigned int>(0, 100);
+                for (unsigned int i = 0; i < recipients_size; ++i) {
+                    recipients.push_back({/*scriptPubKey=*/ConsumeScript(fuzzed_data_provider),
+                                        /*nAmount=*/ConsumeMoney(fuzzed_data_provider), 
+                                        /*fSubtractFeeFromAmount=*/fuzzed_data_provider.ConsumeBool()});
+                }
+                // Equally likely to chose one of the three values.
+                std::vector<int> random_positions = {-1, fuzzed_data_provider.ConsumeIntegralInRange<int>(0, recipients_size), fuzzed_data_provider.ConsumeIntegral<int>()};
+                int change_pos = PickValue(fuzzed_data_provider, random_positions);
+
+                auto res = CreateTransaction(wallet, recipients, change_pos, coin_control, /*sign=*/fuzzed_data_provider.ConsumeBool());
+                if (!res) return;
+                mtx = CMutableTransaction(*(res->tx));
+            },
+            [&] {
+                // Occasionally, creating a random Transaction for Fuzzing.
+                auto new_mtx = ConsumeDeserializable<CMutableTransaction>(fuzzed_data_provider);
+                if (!new_mtx) return;
+                mtx = *new_mtx;
+            },
+            [&] {
+                LOCK(wallet.cs_wallet);
+                if (fuzzed_data_provider.ConsumeBool()) {
+                    (void)CalculateMaximumSignedTxSize(CTransaction(mtx), &wallet);
+                } else {
+                    (void)CalculateMaximumSignedTxSize(CTransaction(mtx), &wallet, &coin_control);
+                }
+            },
+            [&] {
+                const CTxOut tx_out{ConsumeMoney(fuzzed_data_provider), ConsumeScript(fuzzed_data_provider)}; 
+                CalculateMaximumSignedInputSize(tx_out, &wallet, &coin_control);
+            },
+            [&] {
+                CAmount fee;
+                int random_positions[3] = {-1, fuzzed_data_provider.ConsumeIntegralInRange<int>(0, mtx.vout.size()), fuzzed_data_provider.ConsumeIntegral<int>()};

```

Comment:
I think you could keep same pattern as you had previously:
```cpp
std::vector<int> random_positions = {-1, fuzzed_data_provider.ConsumeIntegralInRange<int>(0, recipients_size), fuzzed_data_provider.ConsumeIntegral<int>()};
int change_pos = PickValue(fuzzed_data_provider, random_positions);
```

instead of 
```cpp
int random_positions[3] = {-1, fuzzed_data_provider.ConsumeIntegralInRange<int>(0, mtx.vout.size()), fuzzed_data_provider.ConsumeIntegral<int>()};
int change_pos = fuzzed_data_provider.PickValueInArray(random_positions);
```